contracts/LifCrowdsale.sol

```diff
@@ -238,13 +231,17 @@ contract LifCrowdsale is Ownable, Pausable {
 
      @param beneficiary Address to which Lif will be sent
      @param weiSent Amount of wei contributed
+     @param rate of tokens per ether at the moment of the contribution
    */
-  function addPrivatePresaleTokens(address beneficiary, uint256 weiSent) onlyOwner {
+  function addPrivatePresaleTokens(
+    address beneficiary, uint256 weiSent, uint256 rate
+  ) onlyOwner {
     require(block.timestamp < startTimestamp);
     require(beneficiary != address(0));
     require(weiSent > 0);
+    require(rate > rate1);
 
-    uint256 tokens = weiSent.mul(privatePresaleRate);
+    uint256 tokens = weiSent.mul(rate);
```

Comment:
Would adding an event here give an easier way to see the details of the presale contributions? Should we add one?

Reply:
we can add one, yes. It will be easier.